controllers/datamover.go

```diff
@@ -1,1073 +1,23 @@
-				return false, fmt.Errorf("volSync operator not found. Please install")
-			}
-
-			return false, err
-		}
-
-		adpNS := corev1.Namespace{}
-		if err := r.Get(r.Context, types.NamespacedName{Name: r.NamespacedName.Namespace}, &adpNS); err != nil {
-			return false, err
-		}
-
-		// add volsync privileged mover annotation
-		err = r.addAnnotations(&adpNS, r.Log)
-		if err != nil {
-			return false, err
-		}
-	}
-
-	dataMoverDeployment := &appsv1.Deployment{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      common.DataMover,
-			Namespace: dpa.Namespace,
-		},
-	}
-
-	if (dpa.Spec.Features == nil) || (dpa.Spec.Features.DataMover == nil) ||
-		(dpa.Spec.Features != nil && dpa.Spec.Features.DataMover != nil && !dpa.Spec.Features.DataMover.Enable) {
-		deleteContext := context.Background()
-		if err := r.Get(deleteContext, types.NamespacedName{
-			Name:      dataMoverDeployment.Name,
-			Namespace: r.NamespacedName.Namespace,
-		}, dataMoverDeployment); err != nil {
-			if k8serror.IsNotFound(err) {
-				return true, nil
-			}
-			return false, err
-		}
-
-		deleteOptionPropagationForeground := metav1.DeletePropagationForeground
-		if err := r.Delete(deleteContext, dataMoverDeployment, &client.DeleteOptions{PropagationPolicy: &deleteOptionPropagationForeground}); err != nil {
-			r.EventRecorder.Event(dataMoverDeployment, corev1.EventTypeNormal, "DeleteDataMoverDeploymentFailed", "Could not delete DataMover deployment:"+err.Error())
-			return false, err
-		}
-		r.EventRecorder.Event(dataMoverDeployment, corev1.EventTypeNormal, "DeletedDataMoverDeploymentDeployment", "DataMover Deployment deleted")
-
-		return true, nil
-	}
-
-	op, err := controllerutil.CreateOrUpdate(r.Context, r.Client, dataMoverDeployment, func() error {
-
-		// Setting Deployment selector if a new object is created as it is immutable
-		if dataMoverDeployment.ObjectMeta.CreationTimestamp.IsZero() {
-			dataMoverDeployment.Spec.Selector = &metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					"component": common.DataMoverController,
-				},
-			}
-		}
-
-		// update the Deployment template
-		err := r.buildDataMoverDeployment(dataMoverDeployment, &dpa)
-		if err != nil {
-			return err
-		}
-
-		// Setting controller owner reference on the dataMover deployment
-		return controllerutil.SetControllerReference(&dpa, dataMoverDeployment, r.Scheme)
-	})
-
-	if err != nil {
-		if k8serror.IsInvalid(err) {
-			cause, isStatusCause := k8serror.StatusCause(err, metav1.CauseTypeFieldValueInvalid)
-			if isStatusCause && cause.Field == "spec.selector" {
-				// recreate deployment
-				// TODO: check for in-progress backup/restore to wait for it to finish
-				log.Info("Found immutable selector from previous deployment, recreating DataMover Deployment")
-				err := r.Delete(r.Context, dataMoverDeployment)
-				if err != nil {
-					return false, err
-				}
-				return r.ReconcileDataMoverController(log)
-			}
-		}
-
+	// When updating from OADP version 1.2.x to 1.3.x, delete
```

Comment:
IMO if we are adding code to delete the volume-snapshot-mover controller deployment then we should also add logic to delete:
- the datamover restic secret created per BSL configured 
- the datamover volume options configmap

Reply:
Did not know about the configmap, can you explain to me how it gets created?

Will check if I can delete the secret

Reply:
checked and the created secrets by code are left in the cluster. Going to write code to remove them

The configmaps I did not see them yet. can you describe how they are created and how they look like?

Reply:
these configmaps, right?
https://github.com/openshift/oadp-operator/blob/master/controllers/datamover.go#L234-L244

Reply:
added code. can you test if the configMaps are deleted? I think they only get created if using `VolumeOptionsForStorageClasses` field in features.DataMover

Reply:
You can add unit tests using https://book.kubebuilder.io/reference/envtest :)
[Example](https://github.com/mateusoliveira43/oadp-operator/blob/c5d06a9f4dfbe43010493b0b35e63e1cdebd5140/pkg/credentials/credentials_test.go#L502)